Modules/EventHub/2.0/deploy.json

```diff
@@ -1,140 +1,324 @@
+      }
+    },
+    "cuaId": {
+      "type": "string",
+      "defaultValue": "",
       "metadata": {
-        "description": "Optional. MessagingUnits for premium namespace"
+        "description": "Optional. Customer Usage Attribution id (GUID). This GUID must be previously registered"
       }
     },
-    "consumerGroupName": {
+    "tagEnvironment": {
       "type": "string",
+      "defaultValue": "",
       "metadata": {
-        "description": "Optional. Name of Consumer Group"
+        "description": "Optional. The name of the Environment"
       }
     },
-    "authorizationRulesRootManageSharedAccessKeyName": {
-      "defaultValue": "RootManageSharedAccessKey",
+    "tagProject": {
       "type": "string",
+      "defaultValue": "",
       "metadata": {
-        "description": "Optional. Authorization Rules Root Managed Shared Access Key."
+        "description": "Optional. The name of the project"
+      }
+    },
+    "tagApplication": {
+      "type": "string",
+      "defaultValue": "",
+      "metadata": {
+        "description": "Optional. The name of the application"
+      }
+    },
+    "tagOwner": {
+      "type": "string",
+      "defaultValue": "",
+      "metadata": {
+        "description": "Optional. The business owner for the application"
+      }
+    },
+    "tagOwnerEmail": {
+      "type": "string",
+      "defaultValue": "",
+      "metadata": {
+        "description": "Optional. The Email address of the business owner for the application"
       }
     }
   },
   "variables": {
+    "namespaceName": "[parameters('namespaceName')]",
+    "namespaceResourceId": "[resourceId('Microsoft.EventHub/Namespaces',parameters('namespaceName'))]",
+    "diagnosticStorageAccountId": "[parameters('diagnosticStorageAccountId')]",
+    "diagnosticWorkspaceId": "[parameters('logAnalyticsWorkspaceId')]",
+    "defaultSASKeyName": "RootManageSharedAccessKey",
+    "authRuleResourceId": "[resourceId('Microsoft.EventHub/namespaces/authorizationRules', parameters('namespaceName'), variables('defaultSASKeyName'))]",
+    "apiVersion": "[providers('Microsoft.EventHub','namespaces').apiVersions[0]]",
+    "pidName": "[concat('pid-', parameters('cuaId'))]"
   },
   "resources": [
     {
-      "apiVersion": "2017-04-01",
-      "name": "[parameters('namespaceName')]",
+      "condition": "[not(empty(parameters('cuaId')))]",
+      "type": "Microsoft.Resources/deployments",
+      "apiVersion": "2018-02-01",
+      "name": "[variables('pidName')]",
+      "properties": {
+        "mode": "Incremental",
+        "template": {
+          "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+          "contentVersion": "1.0.0.0",
+          "resources": []
+        }
+      }
+    },
+    {
       "type": "Microsoft.EventHub/namespaces",
+      "apiVersion": "[variables('apiVersion')]",
+      "name": "[parameters('namespaceName')]",
       "location": "[parameters('location')]",
+      "tags": {
```

Comment:
Same as above, we might need to move this value as an input parameter

Reply:
tags changed to new format